functions.R

```diff
@@ -881,3 +891,150 @@ plotter <- function(df, variable = NULL, show_labels = TRUE){
+  temp_clean$class_of_worker_main_job <- ifelse(grepl('unpaid', temp_clean$class_of_worker_main_job), 
+                                                'Unpaid family work', 
+                                                ifelse(grepl('0|no', temp_clean$class_of_worker_main_job),
+                                                       'Self employed, no employees',
+                                                       ifelse(grepl('/w/', temp_clean$class_of_worker_main_job),
+                                                              'Self employed, with employees',
+                                                              ifelse(grepl('Private', temp_clean$class_of_worker_main_job), 
+                                                                     'Private employee', 
+                                                                     ifelse(grepl('Public', temp_clean$class_of_worker_main_job),
+                                                                            'Publice employee', 'Not applicable')))))
+  
+  temp_clean$industry_of_main_job_naics_200718 <- ifelse(grepl('Utilities|Construction|Manufact', as.character(temp_clean$industry_of_main_job_naics_200718)),
+                                                         'Utilities/Construction/Manufacturing',
+                                                         ifelse(grepl('Trade', as.character(temp_clean$industry_of_main_job_naics_200718)), 
+                                                                'Trade Retail/Whosale',
+                                                                ifelse(grepl('Accomm/Food Services|Other Services', as.character(temp_clean$industry_of_main_job_naics_200718)),
+                                                                       'Accommodation/Food/Other Services',
+                                                                       ifelse(grepl('Agric|Forest', as.character(temp_clean$industry_of_main_job_naics_200718)),
+                                                                              'Agriculture/Forest/Fish/Mine/Oil&Gas', as.character(temp_clean$industry_of_main_job_naics_200718)))))
+  
+  temp_clean$fulltime_or_parttime_main_or_only_job <- ifelse(grepl('Full', temp_clean$fulltime_or_parttime_main_or_only_job), 
+                                                             'Full time (30+)',
+                                                             ifelse(grepl('Part', temp_clean$fulltime_or_parttime_main_or_only_job), 
+                                                                    'Part time (1 to 29 hours)','Not applicable'))
+  
+  # is it better to recode them all or create new variables that are just true of false for type of couple, employeds, kids etc?
+  temp_clean$type_of_economic_family <- as.character(temp_clean$type_of_economic_family)
+  temp_clean$type_of_economic_family <- ifelse(temp_clean$type_of_economic_family == 'H-W:2earn,0 kids<25',
```

Comment:
Would it not be easier to recode this using some regex, rather than ifelse?

For example, replace "H" with "husband", remove "-", replace "W" with "wife", etc?

Reply:
(Doesn't need to change, since this apparently works, but for next time...)